### PR TITLE
CQL Basics Cheatsheet

### PR DESCRIPTION
**Pull request by c-schuler**
A very simple cheatsheet for CQL basics.

### CODE REVIEW

File: Source/CheatSheet/CQLBasics-cheatsheet.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+|Value|Description|Example|
+|-----|-----------|-------|
+|Null|The null literal|<pre lang="cql">null</pre>|
+|Boolean|The boolean literals|<pre lang="cql">true, false</pre>|
+|Integer|Sequences of digits in the range 0..2<sup>31</sup>-1|<pre lang="cql">16, -28</pre>|
+|Long|Sequences of digits in the range 0..2<sup>63</sup>-1|<pre lang="cql">16000000000, -28000000000</pre>|
+|Decimal|Sequences of digits with a decimal point, in the range 0.0.. (10<sup>28</sup>-1)/10<sup>8</sup>|<pre lang="cql">100.015</pre>|
+|String|Strings of any character enclosed within single-ticks (')|<pre lang="cql">'pending', 'active', 'complete'</pre>|
+|Date|The at-symbol (@) followed by an ISO-8601 compliant representation of a date|<pre lang="cql">@2014-01-25</pre>|
+|DateTime|The at-symbol (@) followed by an ISO-8601 compliant representation of a datetime|<pre lang="cql">@2014-01-25T14:30:14.559&#13;@2014-01T</pre>|
```
Comment (jbyrdevans) [inline Source/CheatSheet/CQLBasics-cheatsheet.md:14]: Is `@2014-01T` valid?

File: Source/CheatSheet/CQLBasics-cheatsheet.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+|,|Comma for delimiting items in a syntactic list|
+|= != <= < > >=|Comparison operators for comparing values|
+|+ - * / ^|Arithmetic operators for performing calculations|
+
+## Identifiers
+
+|Type|Description|Example|
+|----|-----------|-------|
+|Simple|Any alphabetical character or an underscore, followed by any number of alpha-numeric characters or underscores|<pre lang="cql">Foo1</pre>|
+|Delimited|any sequence of characters enclosed in backticks (`)|<pre lang="cql">Encounter, Performed</pre>|
```
Comment (jbyrdevans) [inline Source/CheatSheet/CQLBasics-cheatsheet.md:43]: Missing the backticks in the example
```suggestion
|Delimited|any sequence of characters enclosed in backticks (`)|<pre lang="cql">`Encounter, Performed`</pre>|
```

File: Source/CheatSheet/CQLBasics-cheatsheet.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
+# CQL Basics Cheat Sheet
```
Comment (jbyrdevans) [inline Source/CheatSheet/CQLBasics-cheatsheet.md:1]: General comment: I would suggest we add some links to the CQL spec in each of the sections.

File: Source/CheatSheet/CQLBasics-cheatsheet.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+|Interval|Set of values between two boundaries that can be inclusive ([]) or exclusive (())|<pre lang="cql">Interval[3, 5) // 3 and 4, but not 5&#13;Interval[@2014-01-01, @2015-01-01) // same as Interval[@2014-01-01, @2014-12-31]</pre>|
+
+## Symbols
+
+|Symbol|Description|
+|------|-----------|
+|:|Definition operator, typically read as "defined as". Also used to separate the numerator from denominator in Ratio literals|
+|()|Parentheses for delimiting groups, as well as specifying and passing function parameters|
+|[]|Brackets for indexing into lists and strings, as well as delimiting the retrieve expression|
+|{}|Braces for delimiting lists and tuples|
```
Comment (jbyrdevans) [inline Source/CheatSheet/CQLBasics-cheatsheet.md:31]: Should mention that Braces are used for lists too

File: Source/CheatSheet/CQLBasics-cheatsheet.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+|<>|Angle-brackets for delimiting generic types within type specifiers|
+|.|Period for qualifiers and accessors|
+|,|Comma for delimiting items in a syntactic list|
+|= != <= < > >=|Comparison operators for comparing values|
```
Comment (jbyrdevans) [inline Source/CheatSheet/CQLBasics-cheatsheet.md:35]: Add equivalence as well:
```suggestion
|= != <= < > >=|Comparison operators for comparing values|
|~ !~|Equivalence operators for comparing values|
```

File: Source/CheatSheet/CQLBasics-cheatsheet.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+|Time|The at-symbol (@) followed by an ISO-8601 compliant representation of a time|<pre lang="cql">@T12:00&#13;@T14:30:14.559</pre>|
+|Quantity|An integer or decimal literal followed by a datetime precision specifier, or a UCUM unit specifier|<pre lang="cql">6 'gm/cm3'&#13;80 'mm[Hg]'&#13;3 months</pre>|
+|Ratio|A ratio of two quantities, separated by a colon (:)|<pre lang="cql">1:128&#13;5 'mg' : 10 'mL'</pre>|
+|Code|Construct consistent with the way terminologies are typically represented|<pre lang="cql">Code '66071002' from "SNOMED-CT" display 'Type B viral hepatitis'</pre>|
+|Concept|Construct to specify multiple terminologies used to code for the same concept|<pre lang="cql">Concept {&#13;  Code '66071002' from "SNOMED-CT",&#13;  Code 'B18.1' from "ICD-10-CM"&#13;} display 'Type B viral hepatitis'</pre>|
+|Tuple|Structured values that contain named elements, each having a value of some type|<pre lang="cql">define "Info":&#13;  Tuple {&#13;    Name: 'Patrick',&#13;    DOB: @2014-01-01,&#13;    Address: Tuple { Line1: '41 Spinning Ave', City: 'Dayton', State: 'OH' },&#13;    Phones: { Tuple { Number: '202-413-1234', Use: 'Home' } }&#13;  }</pre>|
```
Comment (jbyrdevans) [inline Source/CheatSheet/CQLBasics-cheatsheet.md:20]: Might be helpful to point out that you can create Tuples that represent a FHIR resource:
```
FHIR.Extension {
    "url": FHIR.uri { "value": 'some-uri' },
    "value": FHIR.code {
      "value": 'some value'
    }
  }
```

File: Source/CheatSheet/CQLBasics-cheatsheet.md (hunk 1 above)
Comment (c-schuler) [reply Source/CheatSheet/CQLBasics-cheatsheet.md:14]: Yes. I took the examples directly from the spec: https://cql.hl7.org/02-authorsguide.html#simple-values

File: Source/CheatSheet/CQLBasics-cheatsheet.md (hunk 4 above)
Comment (c-schuler) [reply Source/CheatSheet/CQLBasics-cheatsheet.md:31]: It does mention lists, right?

File: Source/CheatSheet/CQLBasics-cheatsheet.md (hunk 5 above)
Comment (c-schuler) [reply Source/CheatSheet/CQLBasics-cheatsheet.md:35]: Interesting. Yeah, it is odd that equivalence is not included in the symbol table: https://cql.hl7.org/03-developersguide.html#symbols. May also be an issue with the spec.

File: Source/CheatSheet/CQLBasics-cheatsheet.md (hunk 6 above)
Comment (c-schuler) [reply Source/CheatSheet/CQLBasics-cheatsheet.md:20]: I figured that would be good to include in the advanced cheat sheet. Along with multi-source queries, let statements, etc...

File: Source/CheatSheet/CQLBasics-cheatsheet.md (hunk 3 above)
Comment (c-schuler) [reply Source/CheatSheet/CQLBasics-cheatsheet.md:1]: Applied

**Pull request merged**